contracts/libs/0.8.x/BytecodeStorageV1.sol

```diff
@@ -141,26 +180,28 @@ library BytecodeStorage {
             // store length of data in first 32 bytes
             mstore(data, size)
             // copy code to memory, excluding the deployer-address
-            extcodecopy(_address, add(data, 0x20), DATA_OFFSET, size)
+            extcodecopy(_address, add(data, 0x20), dataOffset, size)
         }
     }
 
     /**
      * @notice Get address for deployer for given contract bytecode
-     * @param _address address of deployed contract with bytecode containing concat(invalid opcode,  deployer-address, data)
+     * @param _address address of deployed contract with bytecode containing concat(invalid opcode, version, deployer-address, data)
      * @return writerAddress address read from contract bytecode
      */
     function getWriterAddressForBytecode(
         address _address
     ) internal view returns (address) {
         // get the size of the data
         uint256 bytecodeSize = _bytecodeSizeAt(_address);
-        // handle case where address contains code < DATA_OFFSET
+        // the dataOffset for the bytecode
+        uint256 addressOffset = _bytecodeAddressOffsetAt(_address);
+        // handle case where address contains code < addressOffset
         // note: the first check here also captures the case where
         //       (bytecodeSize == 0) implicitly, but we add the second check of
         //       (bytecodeSize == 0) as a fall-through that will never execute
-        //       unless `DATA_OFFSET` is set to 0 at some point.
-        if ((bytecodeSize < DATA_OFFSET) || (bytecodeSize == 0)) {
+        //       unless `addressOffset` is set to 0 at some point.
+        if ((bytecodeSize < addressOffset) || (bytecodeSize == 0)) {
             revert("ContractAsStorage: Read Error");
         }
```

Comment:
I think this case is never possible?
we revert if bytecode size is zero in `_bytecodeVersionAt` (called within `_bytecodeAddressOffsetAt`), and I don't think it is possible for bytecodeSize to be less than the addressOffset unless we messed up the values of the constants in this lib.
I do think we might want to revert if `bytecodeSize < addressOffset + 32` though?

Reply:
++; SGTM, will update to just check `bytecodeSize < addressOffset + 32`